### ISSUE TITLE
Tao datum `spin_map_ptc` does not update after changing lattice parameters

### ISSUE THREAD

**Issue by eiad-hamwi** [bmad-sim/bmad-ecosystem#1259]
As the title describes, the `spin_map_ptc.NNNNNN` datum does not update given any change in the lattice parameters, once the lattice has been initialized. The same is true for `show spin -spin_tune`.

I've uploaded a sample lattice with that as the only datum to try out. Consider changing the lattice energy to see how the datum does not change: 
```
show val data::spin.adst|model
set ele 0 e_tot = 100e9
show val data::spin.adst|model
```

[hsr_high_energy.zip](https://github.com/user-attachments/files/17579006/hsr_high_energy.zip)

**Comment by DavidSagan** [comment]
A prior code change seems to have fixed this.

**Comment by eiad-hamwi** [comment]
Indeed, the issue is fixed for the prior lattice. However, in the following lattice the issue remains. 

[test_lat.zip](https://github.com/user-attachments/files/18028790/test_lat.zip)

A large difference between the two is the presence of several custom overlays. However, removing all the overlays does not fix the issue.

**Comment by DavidSagan** [comment]
Fixed.

**Comment by eiad-hamwi** [comment]
It seems like Tao will crash if attempting to change the energy of more than 1 universe when the `spin_map_ptc` is a datum.

The following tao.init file calls two universes, and has the `spin_map_ptc` datum.
[multi_universe_test.zip](https://github.com/user-attachments/files/18543886/multi_universe_test.zip)

After running Tao, change the energy of both universes:
```
set ele 1@0 e_tot = 100e9
set ele 2@0 e_tot = 110e9
```

Tao should crash after the second command. However, if you remove the `spin_map_ptc` datum, then this does not cause Tao to crash.

**Comment by DavidSagan** [comment]
Fixed. If there are any more issues, please open a new Issue.
